### DISCUSSION TITLE
Validation nested fields with required_if or required_with

### DISCUSSION THREAD

**Discussion by t-prod** [livewire/livewire#5299] [1. Help] [answered]
Hi,

I try to figure out how can I validate nested fields like this but it doesn't work at all.

`'lines.*.antiwaste_department' => 'nullable|required_if:lines.*.antiwaste_discount,1',`

The rule required_if does not seem to get the value of the nested field here.

Have any ideas ?

Regards

**Comment by ridxery** [comment, marked as answer]
'lines.*.antiwaste_department' => 'nullable|required_if:lines.**0**.antiwaste_discount,1', , 

you need to specify which data to get by providing array key/index instead of *, like above.

FYI, when you create a rule with wildcard * even if one of the items validation triggers (fails) all of the items error messages will be shown even on the ones it has correct validated data.


```php
foreach ($this->invoiceItems ?? [] as $key => $item)
        {
            $rules['invoiceItems.'.$key.'.name'] = 'required|string|min:2|max:250';
            $rules['invoiceItems.'.$key.'.price'] = 'required|numeric|min:1';
            $rules['invoiceItems.'.$key.'.quantity'] = 'required|numeric|min:1';
            $rules['invoiceItems.'.$key.'.total_amount'] = 'required|numeric|min:0';
        }
```

**Reply by t-prod** [reply]
@ridxery it's perfect thanks for your feedback. I must use the index to do so ;)
